src/DDTrace/Util/HexConversion.php

```diff
@@ -6,6 +6,29 @@ final class HexConversion
 {
     public static function idToHex($id)
     {
-        return str_pad(dechex($id), 16, "0", STR_PAD_LEFT);
+        $hex = dd_trace_dec_hex((string) $id);
+        return str_pad($hex, 16, "0", STR_PAD_LEFT);
+    }
+
+    public static function hexToInt($hex)
+    {
+        if (strlen($hex) > 16) {
+            throw new \InvalidArgumentException(
+                '\HexConversion::hexToInt() arg must be hexstring with <= 16 characters'
+            );
+        }
+
+        if (strlen($hex) == 16 && strpos("89abcdef", $hex[0]) !== false) {
+            // over PHP_INT_MAX, so needs to be shifted right
+            // per original dd_trace_generate_id implementation.
+            $split = str_split($hex);
+            $hex = "";
+            foreach ($split as $char) {
+                $hex .= base_convert($char, 16, 2);
```

Comment:
I am a bit confused. This seems to be converting from hex digits to binary digits. Is that correct? If so I am not sure I understand what this function is doing. It appears to convert from hex to binary, then inputs the binary representation (named `$hex`?) to function `dd_trace_hex_dec` that supposedly expects a hex representation?
Please clarify.

Reply:
This is `>> 1` https://github.com/signalfx/signalfx-php-tracing/blob/1c349b93875c6ff6362b24d7509539883cd9ac95/src/ext/random.c#L20, which afaict isn't possible without base conversion since there is no unsigned int in php.

Reply:
@tigrannajaryan, I see the mistake on my part.  Correcting now.